src/WebJobs.Extensions/Extensions/Timers/Listener/TimerListener.cs

```diff
@@ -241,6 +235,21 @@ private void OnTimer(object sender, ElapsedEventArgs e)
             }
         }
 
+        private void StartTimer(DateTime now)
+        {
+            // We need to calculate the next interval based on the current
+            // time as we don't know how long the previous function invocation took.
+            // Example: if you have an hourly timer invoked at 12:00 and the invocation takes 1 minute,
+            // we want to calculate the interval for the next timer using 12:01 rather than at 12:00.
+            // Otherwise, you'd start a 1-hour timer at 12:01 when we really want it to be a 59-minute timer.
+
+            // Recalculate the next occurrence as well. For short (1-second) intervals, it's possible that we've 
+            // missed the next invocation and do not want to end up with a negative interval.
+            ScheduleStatus.Next = _schedule.GetNextOccurrence(now);
```

Comment:
InvokeJobFunction is also updating ScheduleStatus.Next and persisting it to storage. So it's confusing to me that we'd recompute it again here. Is the situation you're talking about a scenario where the interval is so short that the time it takes us to persist it to storage causes us to miss the next expected? 

Perhaps what we should do is have the existing StartTimer(Interval) function interpret negative intervals as zero. That would handle this corner case, causing the function to run immediately.